### PR TITLE
breaking.md cleanup - release/v2int/1.2

### PR DESCRIPTION
**Pull request by scottn12**
Cleanup breaking.md

Context: https://github.com/microsoft/FluidFramework/pull/12161

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -17,18 +15,23 @@ It's important to communicate breaking changes to our stakeholders. To write a g
 - Avoid using code formatting in the title (it's fine to use in the body).
 - To explain the benefit of your change, use the [What's New](https://fluidframework.com/docs/updates/v1.0.0/) section on FluidFramework.com.
 
-# 3.0.0
+# 2.0.0-internal-1.1.0
```
Comment (markfields) [inline BREAKING.md:18]: Isn't it `internal.1.1`, not `internal-1.1`?

File: BREAKING.md [hunk 2]
```diff
@@ -17,18 +15,23 @@ It's important to communicate breaking changes to our stakeholders. To write a g
...
-- [Remove `ShareLinkTypes` interface](#Remove-ShareLinkTypes-interface)
-- [Remove `enableShareLinkWithCreate` from `HostStoragePolicy`](#Remove-enableShareLinkWithCreate-from-HostStoragePolicy)
+## 2.0.0-internal-1.1.0 Upcoming changes
 - [Add assertion that prevents sending op while processing another op](#add-assertion-that-prevents-sending-op-while-processing-another-op)
--
-### Remove `type` field from `ShareLinkInfoType`
+- [Remove type field from ShareLinkInfoType](#Remove-type-field-from-ShareLinkInfoType)
+- [Remove ShareLinkTypes interface](#Remove-ShareLinkTypes-interface)
+- [Remove enableShareLinkWithCreate from HostStoragePolicy](#Remove-enableShareLinkWithCreate-from-HostStoragePolicy)
+- [Various return types in @fluidframework/sequence have been widened to include undefined](#various-return-types-in-fluidframeworksequence-have-been-widened-to-include-undefined)
```
Comment (markfields) [inline BREAKING.md:25]: This looks new - I'd pull that into a separate PR after this cleanup

File: BREAKING.md (hunk 1 above)
Comment (markfields) [reply BREAKING.md:18]: Also... should this section (for `next` stuff) be about `2.0.0-internal-2.0.0`?  i.e. the next major bump?

File: BREAKING.md [hunk 3]
```diff
@@ -1,10 +1,8 @@
...
 
-Notes on breaking, upcoming, and otherwise interesting changes go here.  They will be reviewed and published along with each release.  Published changelogs may be found on the docs site at fluidframework.com.
+Notes on breaking, upcoming, and otherwise interesting changes go here. They will be reviewed and published along with each release.  Published changelogs may be found on the docs site at fluidframework.com.
 
-Upcoming changes include anything expected to become a breaking change in the future.  It can include deprecations, optional to required transitions, etc.  They should be added to the section for the version in which they are being announced.
-
-Breaking changes include anything that a consumer upgrading to the specified version must account for as part of the upgrade process.  It can include expected compile time breaks, runtime compatibility breaks, etc.  They should typically be announced as an upcoming change in an earlier version before becoming a breaking change.
+For more information on best practices around communicating breaking please see our docs [here](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes)
```
Comment (markfields) [inline BREAKING.md:5]: nit to sound more authoritative:
```suggestion
For instructions on how to communicate breaking changes please see our docs [here](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes)
```

File: BREAKING.md (hunk 1 above)
Comment (scottn12) [reply BREAKING.md:18]: > Also... should this section (for `next` stuff) be about `2.0.0-internal-2.0.0`? i.e. the next major bump?

Yes, everything under the `2.0.0-internal-1.0.0 Upcoming changes` should be actionable items that you can take for to prepare for breaking changes in a future major release

[More in-depth explanation](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes#breaking-changes-vs-upcoming-changes)

File: BREAKING.md [hunk 4]
```diff
@@ -40,24 +43,47 @@ This field has been deprecated and will be removed in a future breaking change.
...
-# 2.0.0
 
-## 2.0.0 Upcoming changes
-- [Remove `documentId` field from `MockFluidDataStoreContext`](#Remove-documentId-field-from-MockFluidDataStoreContext)
-- [Narrow type of `clientId` field on `MockFluidDataStoreRuntime`](#Narrow-type-of-clientId-field-on-MockFluidDataStoreRuntime)
-- [Remove `ConnectionState.Connecting`](#Remove-ConnectionState.Connecting)
-- [`IContainerRuntime.flush` is deprecated](#icontainerruntimeflush-is-deprecated)
+# 2.0.0-internal-1.0.0
+
+## 2.0.0-internal-1.0.0 Upcoming changes
```
Comment (markfields) [inline BREAKING.md:51]: For future consideration: I'm doing some brainstorming about how the Upcoming changes are organized/presented.

1. Do we want the "upcoming" section to be tied to a version?
    * Pro of mentioning a version is it may be required to be on at least that version to get the alternative to migrate to.  If this is the common/typical case, then probably leave it.
    * Con is that this information gets spread throughout this file, which may make it harder to keep up with the list.  I'm also confused about if there'd be Upcoming sections for minor releases - looks like there are.

2. In light of this realization, maybe we want a separate file, so this info is not interleaved with true Breaking changes, especially since it's updated on a different cadence.  And I imagine there could also be weirdness keeping Upcoming change notes in sync between main and next, whereas that wouldn't so much be the case with Breaking (which would only be added to next, right?).

Comment (markfields) [reply BREAKING.md:51]: cc @scottn12

File: BREAKING.md [hunk 5]
```diff
@@ -334,8 +335,6 @@ Additionally, please note that the `Connecting` state is being renamed to `Catch
 `ConnectionState.Connecting` is marked as deprecated, please use `ConnectionState.CatchingUp` instead.
 `ConnectionState.Connecting` will be removed in the following major release.
 
-### Remove ICodeLoader from `@fluidframework/container-definitions`
```
Comment (markfields) [inline BREAKING.md:337]: This is suspicious... Did we not do what this says?  If it was reverted I would leave it in with a big bold note saying it was reverted.  Maybe strikethrough the text (I did this with bindToContext last week)

File: BREAKING.md (hunk 1 above)
Comment (markfields) [reply BREAKING.md:18]: Ok I see now that minor bumps get Upcoming sections.  I would lean towards using the major version in the H1 heading.

Or maybe `# 2.0.0-internal.1.x` or `# 2.0.0-internal.1.0.0 through 2.0.0-internal.1.1.0` and update it each release.  I'd prefer the former (`1.x`)

**Review by markfields — COMMENTED** [review]
Really glad to see this PR!

File: BREAKING.md (hunk 5 above)
Comment (scottn12) [reply BREAKING.md:337]: This note was actually moved, not deleted. It was originally placed into the wrong section and was moved into the correct one in this PR. You can find this note on line 141 now

**Review by ssimic2 — APPROVED** [review]

**Comment by ssimic2** [comment]
Good feedback @markfields. Let's separate improvements into follow up work, since we need to release 1.2 patch to unblock partners and share with them our (now) more coherent breaking docs.
